Modules/Filtering/FFT/CMakeLists.txt

```diff
@@ -6,5 +6,16 @@ else()
   set(ITKFFT_LIBRARIES ITKFFT)
 endif()
 set(ITKFFT_SYSTEM_LIBRARY_DIRS ${FFTW_LIBDIR})
-set(ITKFFT_INCLUDE_DIRS ${FFTW_INCLUDE})
+set(ITKFFT_INCLUDE_DIRS "${FFTW_INCLUDE}")
+
+if(NOT ITK_NO_FFT_FACTORY_REGISTER_MANAGER)
+  list(APPEND ITKFFT_INCLUDE_DIRS "${CMAKE_BINARY_DIR}/Wrapping/ITKFactoryRegistration")
+
+  # FIXME compile definitions are not propagating to test driver project as expected
+  # add_compile_definitions(ITK_FFT_FACTORY_REGISTER_MANAGER)
```

Comment:
Expected `ITK_FFT_FACTORY_REGISTER_MANAGER` to propagate to the `test` subdirectory from here, but it does not appear in the subdirectory list of directory compile definitions and is not defined in test driver sources unless included directly in `FFT/test/CMakeLists.txt`. Where should variables such as this be defined in the ITK CMake architecture so that they propagate?

Reply:
@dzenanz 
> If you want something accessible in bottom scope, you should define in in parent scope before entering that subdirectory.

Agreed, I expected this to define compile definitions for all projects in the `FFT` directory and then enter subdirectories with `itk_module_impl()`. The fact that this doesn't occur seems to imply that the module-level `CMakeLists.txt` file (i.e. `Modules/Filtering/FFT`) is being handled after its subdirectories (i.e. `Modules/Filtering/FFT/test`). If that's the case, is there another CMake file where definitions should be added in order to propagate into subdirectories? Otherwise, are there other flags or properties that we need to set in order for compiler definitions to propagate from this `CMakeLists.txt`?

Reply:
Ah okay, thanks for the explanation! I didn't understand that library modules and tests are compiled in separate stages. I'll take another look at this and the image IO classes to try to understand the best way to include definitions in tests / library consumers.

Reply:
Resolved: Tests need not rely on default object factory registrations processed in `UseITK.cmake` and should manually register their default factories. CMake definitions are not needed here.